src/app/_services/api/question.service.ts

```diff
@@ -5,6 +5,10 @@ import {APIResponse, Question} from '@app/_models';
 import {PaginatedResult} from '@app/_models/paginatedResult';
 import {catchError} from 'rxjs/operators';
 import {ApiService} from "@app/_services/api.service";
+import {McqRequestData} from "@app/problems/_models/mcq/mcq-request-data";
+import {JavaRequestData} from "@app/problems/_models/java/java-request-data";
+import {ParsonsRequestData} from "@app/problems/_models/parsons/parsons-request-data";
+import {ProblemSetOptions} from "@app/problems/_models/problem-set-options";
```

Comment:
These are not models. These are form data. You should keep them in the corresponding forms. Please also keep the naming conventions `XxxFormData`. for example, `ProblemSetFilterFormData` would be a better name.